src/resultsdisassemblypage.cpp

```diff
@@ -0,0 +1,159 @@
+    ui->setupUi(this);
+    ui->splitter->setStretchFactor(0, 1);
+    ui->splitter->setStretchFactor(1, 1);
+    ui->splitter->setStretchFactor(2, 8);
+}
+
+ResultsDisassemblyPage::~ResultsDisassemblyPage() = default;
+
+/**
+ *  Clear
+ */
+void ResultsDisassemblyPage::clear() {
+    if (ui->asmView->model() == nullptr) return;
+
+    int rowCount = ui->asmView->model()->rowCount();
+    if (rowCount > 0) {
+        ui->asmView->model()->removeRows(0, rowCount, QModelIndex());
+    }
+}
+
+/**
+ *  Produce and show disassembly with 'objdump'
+ */
+void ResultsDisassemblyPage::showDisassembly() {
+    showDisassemblyByAddressRange();
+}
+
+/**
+ *  Produce disassembler with 'objdump' by addresses range and output to Disassembly tab
+ */
+void ResultsDisassemblyPage::showDisassemblyByAddressRange() {
+    // Show empty tab when selected symbol is not valid
+    if (m_curSymbol.symbol.isEmpty()) {
+        clear();
+        return;
+    }
+
+    // Call objdump with arguments: addresses range and binary file
+    QString processName =
+            m_objdump + QLatin1String(" -d --start-address=0x") + QString::number(m_curSymbol.relAddr, 16) +
+            QLatin1String(" --stop-address=0x") +
+            QString::number(m_curSymbol.relAddr + m_curSymbol.size, 16) +
+            QLatin1String(" ") + m_curSymbol.path;
+
+    showDisassembly(processName);
+}
+
+/**
+ *  Produce disassembler with 'objdump' and output to Disassembly tab
+ */
+void ResultsDisassemblyPage::showDisassembly(QString processName) {
+    QTemporaryFile m_tmpFile;
+    QProcess asmProcess;
+
+    if (m_tmpFile.open()) {
+        asmProcess.start(processName);
+
+        if (!asmProcess.waitForStarted() || !asmProcess.waitForFinished()) {
+            return;
+        }
+        QTextStream stream(&m_tmpFile);
+        stream << asmProcess.readAllStandardOutput();
+        m_tmpFile.close();
+    }
+
+    if (m_tmpFile.open()) {
+        int row = 0;
+        QStandardItemModel *model = new QStandardItemModel();
+
+        QStringList headerList;
+        headerList.append(QLatin1String("Assembly"));
+        model->setHorizontalHeaderLabels(headerList);
+
+        QTextStream stream(&m_tmpFile);
+        while (!stream.atEnd()) {
+            QString asmLine = stream.readLine();
+            if (asmLine.isEmpty() || asmLine.startsWith(QLatin1String("Disassembly"))) continue;
+
+            QStandardItem *asmItem = new QStandardItem();
+            asmItem->setText(asmLine);
+            model->setItem(row, 0, asmItem);
+            row++;
+        }
+        ui->asmView->setModel(model);
+    }
+}
+
+/**
+ * Set current chosen function symbol for Disassembler
+ */
+void ResultsDisassemblyPage::setData(const Data::Symbol &symbol) {
+    m_curSymbol = symbol;
+
+    if (m_curSymbol.symbol.isEmpty()) {
+        return;
+    }
+    // If binary is not found at the specified path, use current binary file located at the application path
```

Comment:
all of this mapping logic looks like it should be done in perfparser instead, it knows where a file resides after all (that's where it parses the debug info from e.g.)

Reply:
How do you suggest to pass it to hotspot from perfparser? Enter additional Data::Symbol field for actual path?  Probably it is better to compute it here in hotspot? Do you insist I move the mapping into perfparser?

Reply:
yes, ideally we add another path to the symbol with the path to the mapped file

and yes, I insist on that, as it would be required for anyone else consuming the new perfparser feature to integrate a dissasembly viewer